addons/controllers/kapp-controller/kappcontrollerconfig_controller_test.go

```diff
@@ -0,0 +1,150 @@
+		f, err := os.Open(clusterResourceFilePath)
+		Expect(err).ToNot(HaveOccurred())
+		defer f.Close()
+		err = testutil.CreateResources(f, cfg, dynamicClient)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		By("Deletingcluster and KappControllerConfig")
+		f, err := os.Open(clusterResourceFilePath)
+		Expect(err).ToNot(HaveOccurred())
+		defer f.Close()
+		testutil.DeleteResources(f, cfg, dynamicClient, true)
+	})
+
+	Context("reconcile KappControllerConfig for management cluster", func() {
+
+		BeforeEach(func() {
+			clusterName = "test-cluster-1"
+			clusterResourceFilePath = "testdata/test-1.yaml"
+		})
+
+		It("Should reconcile KappControllerConfig and create data value secret on management cluster", func() {
+
+			key := client.ObjectKey{
+				Namespace: "default",
+				Name:      "test-cluster-1",
+			}
+
+			Eventually(func() bool {
+				cluster := &clusterapiv1beta1.Cluster{}
+				err := k8sClient.Get(ctx, key, cluster)
+				if err != nil {
+					return false
+				}
+				return true
+			}, waitTimeout, pollingInterval).Should(BeTrue())
+
+			Eventually(func() bool {
+				config := &runv1alpha3.KappControllerConfig{}
+				err := k8sClient.Get(ctx, key, config)
+				if err != nil {
+					return false
+				}
+				Expect(config.Spec.Namespace).Should(Equal("test-ns"))
+				//Expect(config.Spec.CreateNamespace).Should(Equal(false))
+				return true
+			}, waitTimeout, pollingInterval).Should(BeTrue())
+
+			Eventually(func() bool {
+				secretKey := client.ObjectKey{
+					Namespace: "default",
+					Name:      util.GenerateDataValueSecretNameFromAddonNames(clusterName, KappControllerAddonName),
+				}
+				secret := &v1.Secret{}
+				err := k8sClient.Get(ctx, secretKey, secret)
+				if err != nil {
+					return false
+				}
+				Expect(secret.Type).Should(Equal(v1.SecretTypeOpaque))
+				secretData := string(secret.Data["values.yaml"])
+				Expect(strings.Contains(secretData, "namespace: test-ns")).Should(BeTrue())
+
+				// TODO: Add more value validation testcases
+				// TODO: Check coreDNSIP value
+				return true
+			}, waitTimeout, pollingInterval).Should(BeTrue())
+
+		})
+
+		// TODO: Check status.secretRef after reconciliation
+
+		It("Should reconcile KappControllerConfig deletion in management cluster", func() {
+
+			key := client.ObjectKey{
+				Namespace: "default",
+				Name:      "test-cluster-1",
+			}
+
+			Eventually(func() bool {
+				config := &runv1alpha3.KappControllerConfig{}
+				err := k8sClient.Get(ctx, key, config)
+				if err != nil {
+					if errors.IsNotFound(err) {
+						return true
+					}
+					return false
+				}
+				err = k8sClient.Delete(ctx, config)
+				if err != nil {
+					return false
+				}
+				return true
+			}, waitTimeout, pollingInterval).Should(BeTrue())
+
+			Eventually(func() bool {
+				secretKey := client.ObjectKey{
+					Namespace: "default",
+					Name:      fmt.Sprintf("%s-secretgen-controller-data-values", clusterName),
```

Comment:
Change addon name here from secretgen-controller to kapp-controller
could be changed to => fmt.Sprintf("%s-%s-data-values", clusterName, constants.KappControllerAddonName

Reply:
Nice catch! Fixed.